x-pack/plugins/actions/server/builtin_action_types/slack.ts

```diff
@@ -127,18 +126,21 @@ async function slackExecutor(
   const { webhookUrl } = secrets;
   const { message } = params;
 
-  let proxyAgent: HttpsProxyAgent | HttpProxyAgent | undefined;
+  let httpProxyAgent: Agent | undefined;
   if (execOptions.proxySettings) {
-    proxyAgent = getProxyAgent(execOptions.proxySettings, logger);
+    const httpProxyAgents = getProxyAgents(execOptions.proxySettings, logger);
+    httpProxyAgent = webhookUrl.startsWith('https')
```

Comment:
**nit:** should we lower case before calling `startsWith`? In case the end user decided to input a webhook url in all caps.

```suggestion
    httpProxyAgent = webhookUrl.toLowerCase().startsWith('https')
```

Reply:
we can do that; not sure `HTTP` and `HTTPS` are valid protocols as urls, but I'm sure it happens in practice :-)